cdap-app-fabric/src/main/java/co/cask/cdap/security/TokenSecureStoreUpdater.java

```diff
@@ -90,6 +98,10 @@ private Credentials refreshCredentials() {
         JobHistoryServerTokenUtils.obtainToken(hConf, refreshedCredentials);
       }
 
+      if (supportsKMS) {
```

Comment:
If you do it using the interface way as mentioned above, you don't need this check at all, you can just do the `instanceof` check here and call the `addDelegationTokens` method from here as well (meaning one less Util class, yeah).